send-transaction-service/src/transaction_client.rs

```diff
@@ -227,37 +227,15 @@ where
 /// * Update the validator identity keypair and propagate the changes to the
 ///   scheduler. Most of the complexity of this structure arises from this
 ///   functionality.
+#[derive(Clone)]
 pub struct TpuClientNextClient {
     runtime_handle: Handle,
     sender: mpsc::Sender<TransactionBatch>,
-    bind_socket: UdpSocket,
-    // This handle is needed to implement `NotifyKeyUpdate` trait. It's only
-    // method takes &self and thus we need to wrap with Mutex.
-    join_and_cancel: Arc<Mutex<(Option<TpuClientJoinHandle>, CancellationToken)>>,
-    leader_forward_count: u64,
-}
-
-// Implement Clone manually because `UdpSocket` implements only `try_clone`.
-impl Clone for TpuClientNextClient {
-    fn clone(&self) -> Self {
-        let bind_socket = self
-            .bind_socket
-            .try_clone()
-            .expect("Cloning bind socket should always finish successfully.");
-
-        TpuClientNextClient {
-            runtime_handle: self.runtime_handle.clone(),
-            sender: self.sender.clone(),
-            bind_socket,
-            join_and_cancel: self.join_and_cancel.clone(),
-            leader_forward_count: self.leader_forward_count,
-        }
-    }
+    update_certificate_sender: watch::Sender<Option<StakeIdentity>>,
+    #[cfg(any(test, feature = "dev-context-only-utils"))]
+    cancel: CancellationToken,
```

Comment:
It would be great to add a note here, explaining what functionality the test require, such that they need access to this cancellation token.

I can see that it is used in `test_utils.rs` by the `Stopped` trait, providing a `stop()` method to the client.
Is there a particular reason this functionality is only useful in test code?